### ISSUE TITLE
Chart is not rendering well with custom colors after data update.

### ISSUE THREAD

**Issue by julb** [jtblin/angular-chart.js#51]
Hi,

I have implemented a pie chart on my page and set custom colours using "colours" attribute.
On first load, eveything is well, but if I update data, the fill color change and isn't consistent.

The behaviour is OK if I do not specify custom colours.

**Comment by jtblin** [comment]
Could you please post a jsbin or similar? It is very hard to investigate issues without repro cases. Here is a [jsbin template](http://jsbin.com/dufibi/3/edit?html,js,output) for convenience.

**Comment by julb** [comment]
Hey jtblin,

Of course! Thanks for the jsbin.

[Here it is.](http://jsbin.com/kalibozaso/1/edit)

You will see that:
- On load, everything is OK
- Click on refresh
- The fill color changes and isn't OK

For information, I'm using Chrome 40.

Regards,

Julien

**Comment by jtblin** [comment]
Thanks for the repro case, I'll look into that later.

**Comment by jtblin** [comment]
I think this is related to #50 Chart.js does not use the different nuances of colours consistently, e.g. `fillColor` is not always used for `fillColor` (sic) and during the `update` I am assigning properties one for one i.e.

```
    item.fillColor = colour.fillColor;
    item.highlightColor = colour.highlightColor;
    item.strokeColor = colour.strokeColor;
    item.pointColor = colour.pointColor;
    item.pointStrokeColor = colour.pointStrokeColor;
```

One solution would be to maintain a lookup table for all charts to know which property is used where but that really seams like a bug that should be fixed in Chart.js itself.

Another solution is to not update the colours when the data changes, and watch the `colours` options and recreate the charts if the colours are updated.

**Comment by jtblin** [comment]
Done and published as `0.4.3`.

**Comment by julb** [comment]
Thanks for your reactivity!
